utils/service_discovery/sd_docker_backend.py

```diff
@@ -171,6 +171,14 @@ def _get_host_address(self, state, c_id, tpl_var):
             if pod_ip:
                 return pod_ip
 
+        # try to get the rancher IP address
+        log.debug("No IP address was found in container %s (%s) "
```

Comment:
Could you add `if Platform.is_rancher()` before this new code? Otherwise this rancher-specific log line will show up in non Rancher environments.

You'll need to create an `is_rancher` function in `platform.py` first. One way you could detect rancher is by looking for their agent container(s) (read more about that [here](https://docs.rancher.com/rancher/v1.6/en/faqs/agents/)).